app/src/main/java/cm/aptoide/pt/notification/NotificationPolicyFactory.java

```diff
@@ -22,7 +26,7 @@ Policy getPolicy(AptoideNotification notification) {
       case AptoideNotification.NEW_FEATURE:
       case AptoideNotification.APPC_PROMOTION:
       case AptoideNotification.CAMPAIGN:
-        return new CampaignPolicy();
+        return new CampaignPolicy(notification.getWhitelistedPackages(), installedApps);
```

Comment:
If I pass the list we have a greater risk of getting an outdated installed apps list which might compromise the feature because we would have to run the installedApps query before.